rbc/omnisci_backend/omnisci_buffer.py

```diff
@@ -347,19 +349,94 @@ def omnisci_buffer_setitem(a, i, v):
 
 
 @extending.intrinsic
-def omnisci_buffer_is_null_(typingctx, data):
-    sig = types.int8(data)
+def omnisci_column_set_null_(typingctx, col_var, row_idx):
+    # Float values are serialized as integers by OmniSciDB
+    # For reference, here is the conversion table for float and double
+    #   FLOAT:  1.1754944e-38            -> 8388608
+    #   DOUBLE: 2.2250738585072014e-308  -> 4503599627370496
+    #                    ^                          ^
+    #                 fp value                  serialized
+    T = col_var.eltype
+    sig = types.void(col_var, row_idx)
+
+    target_info = TargetInfo()
+    null_value = target_info.null_values[str(T)]
 
     def codegen(context, builder, signature, args):
+        zero = int32_t(0)
+
+        data, index = args
+
+        assert data.opname == 'load'
+        buf = data.operands[0]
+
+        ptr = builder.load(builder.gep(buf, [zero, zero]))
+
+        ty = ptr.type.pointee
+        nv = ir.Constant(ir.IntType(T.bitwidth), null_value)
+        if isinstance(T, types.Float):
+            nv = builder.bitcast(nv, ty)
+        builder.store(nv, builder.gep(ptr, [index]))
+
+    return sig, codegen
+
+
+@extending.overload_method(BufferType, 'set_null')
+def omnisci_column_set_null(col_var, index):
+    def impl(col_var, index):
+        return omnisci_column_set_null_(col_var, index)
+    return impl
+
+
+@extending.intrinsic
+def omnisci_column_is_null_(typingctx, col_var, row_idx):
+    T = col_var.eltype
+    sig = types.boolean(col_var, row_idx)
+
+    target_info = TargetInfo()
+    null_value = target_info.null_values[str(T)]
+    nv = ir.Constant(ir.IntType(T.bitwidth), null_value)
 
+    def codegen(context, builder, signature, args):
+        zero = int32_t(0)
+        data, index = args
+        assert data.opname == 'load'
+        buf = data.operands[0]
+
+        ptr = builder.load(builder.gep(buf, [zero, zero]))
+        res = builder.load(builder.gep(ptr, [index]))
+
+        if isinstance(T, types.Float):
+            res = builder.bitcast(res, nv.type)
+
+        return builder.icmp_signed('==', res, nv)
+
+    return sig, codegen
+
+
+@extending.overload_method(BufferType, 'isNull')
+@extending.overload_method(BufferType, 'is_null')
```

Comment:
We should decide which method to support, `isNull` or `is_null`. Not both. See also https://github.com/xnd-project/rbc/issues/240

Reply:
I think C++ `UDTFs` uses `isNull` but I think it makes sense for us to continue using `is_null` to keep consistent with array.

Reply:
Ok, let's use Pythonic `is_null` and remove `isNull` support.